### PR TITLE
Remove usage of distutils

### PR DESCRIPTION
**Pull request by astrofrog**
Just a quick hack to see if this is possible - see some notes below

### CODE REVIEW

File: extension_helpers/_openmp_helpers.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def check_openmp_support(openmp_flags=None):
                 log.warn("Unexpected output from test OpenMP "
                          "program (output was {0})".format(output))
                 is_openmp_supported = False
-        except (CompileError, LinkError, subprocess.CalledProcessError):
+        except Exception:
```
Comment (astrofrog) [inline extension_helpers/_openmp_helpers.py:214]: I can't figure out a way to get compilation specific errors without distutils

File: extension_helpers/_openmp_helpers.py [hunk 2]
```diff
@@ -224,9 +224,10 @@ def is_openmp_supported():
     """
     Determine whether the build compiler has OpenMP support.
     """
-    log_threshold = log.set_threshold(log.FATAL)
+    log_threshold = log.level
+    log.setLevel('CRITICAL')
```
Comment (astrofrog) [inline extension_helpers/_openmp_helpers.py:228]: Not 100% sure if this is the equivalent - setuptools doesn't have a common log like distutils does

File: extension_helpers/tests/__init__.py [hunk 3]
```diff
@@ -83,18 +83,6 @@ def run_setup(setup_script, args):
         raise SystemExit(p.returncode)
 
 
-@pytest.fixture(scope='function', autouse=True)
-def reset_distutils_log():
```
Comment (astrofrog) [inline extension_helpers/tests/__init__.py:87]: Since setuptools doesn't have a common log, not sure we can do this anymore

**Comment by Cadair** [comment]
Well that was less controversial than I was expecting.

File: extension_helpers/_distutils_helpers.py [hunk 4]
```diff
@@ -10,9 +10,10 @@
 
 import os
 import sys
-from distutils import ccompiler
-from distutils.dist import Distribution
-from distutils.errors import DistutilsError
+
+from setuptools.command.build_ext import new_compiler
+from setuptools.dist import Distribution
+from setuptools.errors import DistutilsError
```
Comment (Cadair) [inline extension_helpers/_distutils_helpers.py:17]: I assume none of these imports are new(ish) to setuptools and they have been providing versions of this for ages?

Should we have a test which checks our stuff works with a minimum version of setuptools?

**Review by Cadair — APPROVED** [review]
Other than testing with old setuptools, I can see no problem with this, it seems to work well enough.

File: extension_helpers/_distutils_helpers.py (hunk 4 above)
Comment (astrofrog) [reply extension_helpers/_distutils_helpers.py:17]: Yes sounds good! Will add.

**Pull request merged**